### PR TITLE
Fix minor bugs in the wallet

### PR DESCRIPTION
**Pull request by pedroferreira1**
- Adding timeout to requests;
- Aligning in the center the loading addresses screen;
- If an error happens while loading addresses, the user now can change the server without waiting finish loading (that would never finish).

### CODE REVIEW

File: src/api/axiosInstance.js [hunk 1]
```diff
@@ -17,9 +17,13 @@ import $ from 'jquery';
  *
  * @module Axios
  */
-const createRequestInstance = (resolve) => {
+const createRequestInstance = (resolve, timeout) => {
```
Comment (msbrogli) [inline src/api/axiosInstance.js]: Update the docs. Is it seconds or milliseconds?

File: src/api/wallet.js [hunk 2]
```diff
@@ -45,7 +45,7 @@ const walletApi = {
    */
   sendTokens(txHex, resolve) {
     const postData = {tx_hex: txHex};
-    return createRequestInstance(resolve).post('thin_wallet/send_tokens', postData).then((res) => {
+    return createRequestInstance(resolve, 300000).post('thin_wallet/send_tokens', postData).then((res) => {
```
Comment (msbrogli) [inline src/api/wallet.js]: Maybe it should be a parameter in the config file.

File: src/api/axiosInstance.js [hunk 3]
```diff
@@ -17,9 +17,13 @@ import $ from 'jquery';
...
+  if (timeout === undefined) {
+    timeout = 10000;
```
Comment (msbrogli) [inline src/api/axiosInstance.js]: Maybe it should be a parameter in the config file.

File: src/components/RequestError.js [hunk 4]
```diff
@@ -48,6 +56,9 @@ class RequestErrorModal extends React.Component {
    */
   handleChangeServer = () => {
     $('#requestErrorModal').modal('hide');
+    // When user decides to change the server when an error happened, we must reset the loading information
+    this.props.dataLoaded({addressesFound: 0, transactionsFound: 0});
```
Comment (msbrogli) [inline src/components/RequestError.js]: Shouldn't it be done by the loading screen itself? Maybe in the error callback of a promise?

Comment (pedroferreira1) [reply src/components/RequestError.js]: Hummm, I think you are right. I should do in the reject of the promise that requests the addressesHistory, what do you think? 

Maybe in this line: https://github.com/HathorNetwork/hathor-wallet/blob/master/src/utils/wallet.js#L164

What do you think?

File: src/api/axiosInstance.js (hunk 1 above)
Comment (pedroferreira1) [reply src/api/axiosInstance.js]: Done

File: src/api/axiosInstance.js (hunk 3 above)
Comment (pedroferreira1) [reply src/api/axiosInstance.js]: Done

File: src/api/wallet.js (hunk 2 above)
Comment (pedroferreira1) [reply src/api/wallet.js]: Done

File: src/components/RequestError.js (hunk 4 above)
Comment (msbrogli) [reply src/components/RequestError.js]: I think the best place would be where the transaction is generated. I think that the error screen must show up, and the download of the address history will be restarted.

Maybe in this line: https://github.com/HathorNetwork/hathor-wallet/blob/master/src/utils/wallet.js#L251

What do you think?

Comment (pedroferreira1) [reply src/components/RequestError.js]: You are 100% correct! I was thinking wrongly about the promise chain. Will do this

**Pull request merged**